client/classic/upe/deferred-intent.js

```diff
@@ -41,6 +42,23 @@ jQuery( function ( $ ) {
 		}
 	}
 
+	// Mount the Stripe Payment Elements onto the add payment method page.
+	if (
+		$( 'form#add_payment_method' ).length ||
+		$( 'form#order_review' ).length
+	) {
+		maybeMountStripePaymentElement();
```

Comment:
Something I'm not sure about with this code is the `maybeMountStripePaymentElement` function eventually calls `createStripePaymentElement()` which loads the Stripe PE with the following data:

```
const amount = Number( getStripeServerData()?.cartTotal );
const paymentMethodTypes = getPaymentMethodTypes( paymentMethodType );
const options = {
	mode: amount < 1 ? 'setup' : 'payment',
	currency: getStripeServerData()?.currency.toLowerCase(),
	amount,
	paymentMethodCreation: 'manual',
	paymentMethodTypes,
	appearance: initializeAppearance(),
	fonts: getFontRulesFromPage(),
};
```

Note the `amount` is set to the value of `cartTotal` which is just the WC->cart->total sent in JS params.
This means that if the customer has added a product to their cart, then goes to **My Account > Payment Methods > Add payment method**, this amount will be their cart total and not 0, which means the `options.mode` won't be set to `'setup'`.

From what I could see WooPayments does the same thing and from my testing it doesn't seem to break anything but I just wanted to flag this in case I might be missing something :)

Reply:
Interesting. I didn't find problematic behaviors either, and I see the same on WooPayments, but it feels off to pass inaccurate information there, even though we can't find any unintended effects. 

How about adjusting it to pass the right mode? Glad to do it separately from this PR since this is working as expected and it's how WooPayments handles it. We could also check with the team that implemented this over there.

Reply:
My concern is that if we force the mode to be 'setup' for all add payment method pages, then there might be unintended side effects.

If it's okay with you, I might move this to a seperate issue to investigate further.

Reply:
I've created an issue: https://github.com/woocommerce/woocommerce-gateway-stripe/issues/2798